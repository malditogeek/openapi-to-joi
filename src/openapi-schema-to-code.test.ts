import path from "path"
import openapiSchemaToCode from "./openapi-schema-to-code"

describe("openapiSchemaToCode", () => {
  it("creates expected file", async () => {
    const actual = await openapiSchemaToCode(
      path.join(__dirname, "fixtures", "example1.json")
    )

    expect(actual).toMatchSnapshot()
  })
})
