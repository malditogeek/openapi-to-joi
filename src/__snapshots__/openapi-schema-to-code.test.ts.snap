// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`openapiSchemaToCode creates expected file 1`] = `
"/* eslint-disable */
/* prettier-ignore */
import Joi from \\"joi\\"

export const components = {
  Component1: Joi.object({
    field1: Joi.string().required().email({}),
    field2: Joi.number().required().integer(),
    field3: Joi.number().max(10).min(0).multiple(0.1),
    field4: Joi.string().pattern(
      /^(\\\\d{4})-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])T([01][0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9]|60)(.[0-9]+)?(Z|(\\\\+|-)([01][0-9]|2[0-3]):([0-5][0-9]))$/i,
      { name: \\"JsonSchema date-time format\\" }
    ),
    field5: Joi.string().allow(\\"option1\\", \\"option2\\").only(),
  }).unknown(),
  Component2: Joi.string().allow(\\"option1\\", \\"option2\\").only(),
};
"
`;
